flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -650,10 +651,11 @@ public RelRoot convertQuery(SqlNode query, final boolean needsValidation, final
             result = result.accept(new NestedJsonFunctionRelRewriter());
         }
 
-        // propagate the hints.
-        result = RelOptUtil.propagateRelHints(result, false);
-
         // ----- FLINK MODIFICATION BEGIN -----
+        // propagate the hints.
+        // 'FlinkRelOptUtil.propagateRelHints' will also propagate hints not only in the whole rel
```

Comment:
Nit: `The method FlinkRelOptUtil#propagateRelHints not only finds and propagates hints throughout the entire rel tree but also within subqueries.`